packages/react-native-audio-api/src/plugin/withAudioAPI.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  AndroidConfig,
+  createRunOncePlugin,
+  ConfigPlugin,
+  withInfoPlist,
+  withAndroidManifest,
+} from '@expo/config-plugins';
+
+const pkg = require('react-native-audio-api/package.json');
+
+const withBackgroundAudio: ConfigPlugin = (config) => {
+  return withInfoPlist(config, (iosConfig) => {
+    if (iosConfig.modResults.UIBackgroundModes) {
+      iosConfig.modResults.UIBackgroundModes = [
+        ...Array.from(
+          new Set([...iosConfig.modResults.UIBackgroundModes, 'audio'])
+        ),
+      ];
+    } else {
+      iosConfig.modResults.UIBackgroundModes = ['audio'];
+    }
+    return iosConfig;
+  });
+};
+
+// 'android.permission.FOREGROUND_SERVICE',
+// 'android.permission.FOREGROUND_SERVICE_MEDIA_PLAYBACK',
+
+const withAndroidPermissions: ConfigPlugin<{
+  androidFSPermissions: string[];
+}> = (config, { androidFSPermissions }) => {
+  return AndroidConfig.Permissions.withPermissions(
+    config,
+    androidFSPermissions
+  );
+};
+
+const withForegroundService: ConfigPlugin<{
+  androidFSTypes: string[];
+}> = (config, { androidFSTypes }) => {
+  return withAndroidManifest(config, (mod) => {
+    const manifest = mod.modResults;
+    const mainApplication =
+      AndroidConfig.Manifest.getMainApplicationOrThrow(manifest);
+
+    const SFTypes = androidFSTypes.join('|');
+
+    const serviceElement = {
+      $: {
+        'android:name': 'system.MediaNotificationManager$NotificationService',
+        'android:stopWithTask': 'true',
+        'android:foregroundServiceType': SFTypes,
+      },
+      intentFilter: [],
+    };
+
+    if (!mainApplication.service) {
+      mainApplication.service = [];
+    }
+
+    mainApplication.service.push(serviceElement);
+
+    return mod;
+  });
+};
+
+const withAudioAPI: ConfigPlugin<{
+  iosBackgroundMode: boolean;
+  androidForegroundService: boolean;
```

Comment:
```suggestion
  iosBackgroundMode?: boolean;
  androidForegroundService?: boolean;
```
These should be optional since you are providing default values for them.

Reply:
will do